src/lib/network_pool/transaction_pool.ml

```diff
@@ -323,22 +331,26 @@ struct
               "Couldn't re-add locally generated command $cmd, not valid \
                against new ledger."
               ~metadata:
-                [("cmd", User_command.to_yojson (cmd :> User_command.t))] ;
+                [ ( "cmd"
+                  , User_command.to_yojson (User_command.forget_check cmd) ) ] ;
             remove_cmd ()
           in
           if not (Hashtbl.mem t.locally_generated_committed cmd) then
-            if not (has_sufficient_fee t.pool (cmd :> User_command.t)) then (
+            if not (has_sufficient_fee t.pool (User_command.forget_check cmd))
+            then (
               Logger.info t.logger ~module_:__MODULE__ ~location:__LOC__
                 "Not re-adding locally generated command $cmd to pool, \
                  insufficient fee"
                 ~metadata:
-                  [("cmd", User_command.to_yojson (cmd :> User_command.t))] ;
+                  [ ( "cmd"
+                    , User_command.to_yojson (User_command.forget_check cmd) )
+                  ] ;
               remove_cmd () )
             else
               match
                 Option.bind
-                  (Base_ledger.location_of_key best_tip_ledger
-                     (User_command.sender (cmd :> User_command.t)))
+                  (Base_ledger.location_of_account best_tip_ledger
+                     (User_command.source (User_command.forget_check cmd)))
```

Comment:
This should be the signer account right? In `Indexed_pool.add_from_gossip_exn` 1) we check the nonce against signer nonce. 2) The balance of the signer account should be at least `fee+amount+account_creation_fee` when `signer=source` and `fee+account_creation_fee` when `signer != source`.  (account_creation_fee only if receiver is new)

Reply:
Good catch, fixed.